src/components/timeline/timeline.tsx

```diff
@@ -8,26 +8,57 @@ import { Slider } from "@mui/material";
 import css from "./timeline.scss";
 
 const TICK_COUNT = 16;
+const LOADING_DELAY = 100; // ms
 
 export const Timeline: React.FC = observer(function WrappedComponent() {
-  const { simulation } = useStores();
+  const { simulation, snapshotsManager } = useStores();
+  const timeoutId = useRef(0);
   const [val, setVal] = useState(simulation.timeInYears);
   const [disabled, setDisabled] = useState(true);
+  const [timeProgress, setTimeProgress] = useState("0%");
 
   const tickDiff = simulation.config.simulationEndYear / TICK_COUNT;
   const ticks = Array.from({ length: TICK_COUNT + 1 }, (_, i) => i * tickDiff);
-  const timeProgress = `${simulation.timeInYears / simulation.config.simulationEndYear * 100}%`;
   const marks = ticks.map((tick) => ({ value: tick, label: tick }));
   useEffect(() => {
     setDisabled(!simulation.simulationStarted || (simulation.simulationRunning && !simulation.simulationEnded));
   }, [simulation.simulationStarted, simulation.simulationRunning, simulation.simulationEnded]);
 
   useLayoutEffect(() => {
-    setVal(simulation.timeInYears);
-  }, [simulation.timeInYears]);
+    if (simulation.simulationRunning && snapshotsManager.maxYear > val) {
+      snapshotsManager.restoreLastSnapshot();
+      simulation.start(); //when we restoreLastSnapshot, we need to start the simulation again
+    }
+  },[simulation, simulation.simulationRunning, snapshotsManager, val]);
+
+  useLayoutEffect(() => {
+    if (simulation.simulationRunning) {
+      if (snapshotsManager.maxYear > val) {
+        setVal(snapshotsManager.maxYear);
+        simulation.setTimeInYears(snapshotsManager.maxYear);
+        setTimeProgress(`${snapshotsManager.maxYear / simulation.config.simulationEndYear * 100}%`);
+      }
+    } else if (!simulation.simulationStarted) {
+      setTimeProgress("0%");
+      setVal(0);
+      simulation.setTimeInYears(0);
+    } else {
+      setVal(simulation.timeInYears);
+    }
```

Comment:
Regarding the use of useLayoutEffect - what is the purpose of the first two `if` clauses? The very last `else` clause updates the slider value to the simulation's timeInYears and that's clear. However, I'm unsure about the intentions of the other lines above and whether they are necessary.